src/lib/isLicensePlate.js

```diff
@@ -4,7 +4,7 @@ const validators = {
   'de-LI': str => /^FL[- ]?\d{1,5}[UZ]?$/.test(str),
-  'it-IT': str => /^[ABCDEFGHJKLMNPRSTVWXYZ]{2}\s?(?:\d{5}|\d{3}\s?[ABCDEFGHJKLMNPRSTVWXYZ]{2})$/.test(str),
+  'it-IT': str => /^([ABCDEFGHJKLMNPRSTVWXYZ]{2}\s?\d{3}\s?[ABCDEFGHJKLMNPRSTVWXYZ]{2}|(ROMA|Roma|AG|AL|AN|AO|AP|AQ|AR|AT|AV|BA|BG|BI|BL|BN|BO|BR|BS|BT|BZ|CA|CB|CE|CH|CL|CN|CO|CR|CS|CT|CZ|EN|FC|FE|FG|FI|FM|FR|GE|GO|GR|IM|IS|KR|LC|LE|LI|LO|LT|LU|MB|MC|ME|MI|MN|MO|MS|MT|NA|NO|NU|OR|PA|PC|PD|PE|PG|PI|PN|PO|PR|PT|PU|PV|PZ|RA|RC|RE|RG|RI|RN|RO|SA|SI|SO|SP|SR|SS|SV|TA|TE|TN|TO|TP|TR|TS|TV|UD|VA|VB|VC|VE|VI|VR|VT|VV)\s?\d{1,6})$/.test(str),

```

Comment:
Which period had 1 digit license plate? From what I can see on wikipedia we always had at least 4 digits. 
And to simplify the regex I would split it in two parts so that older license plates are only checked whenever the "actual" regex fails.

Reply:
It depends on the provinces. E.g. for Turin, the plate [TO1](https://www.targheitaliane.com/tabelle-targhe-1927/tabella-completa-1927/?pag=1&tipo=Autoveicolo&prov=TO) was registered on march 1927, and the plate [T09932](https://www.targheitaliane.com/tabelle-targhe-1927/tabella-completa-1927/?pag=16&tipo=Autoveicolo&prov=TO) (the last with 4 digits) on dicember 1927.

Reply:
IMHO we won't make it to support ALL plates, but... hey guys, honestly, is it possible that a 1927 car is still in circulation - _in real life, I mean_ ? :thinking:

Reply:
This is exactly what I meant here https://github.com/validatorjs/validator.js/pull/1677#discussion_r645965273, when I said I preferred the regex with support only from '94 😒

Reply:
Oh sorry, I hadn't seen these comments, they don't appear in the diff tab :smile:

There are more chances to have a '68 car still in circulation than a '27 one anyway :wink: 

And I think support for motorcycle is fubar...

So, here are the options:
- Support for both new and old car plates (even being too permissive for the latter one, eg `<city-codes>\s[A-Z0-9]{5}`)
- Only support cars as of '94, which would result in a really simpler regex...

Reply:
BTW, if we opt for the 1st choice, I like the idea of @fedeci to split it in 2 distinct regexp, like : 
```javascript
var oldPlates = '(ROMA|Roma|AG|AL|AN|AO|AP|AQ|AR|AT...)';
var newPlates = '([ABCDEFGHJKLMNPRSTVWXYZ]{2}...)';
```

I don't know if it's possible we the actual main validator script implementation, though :thinking:

Reply:
Yes it is possibile since it is an arrow function. We should get an implementation similar to this:
```js
'it-IT': str => newPlates.test(str) || oldPlates.test(str),
```

Reply:
> Support for both new and old car plates (even being too permissive for the latter one, eg <city-codes>\s[A-Z0-9]{5})

I'm sure that old plates start from TO1 to TO 999999 and then come's the plates with a letter TO A00000. So the regex would be something like `<city-codes>\s(?:\d{5,6}|[A-Z0-9]{6})` => in `\d{5,6}` the 5 can be whatever you want It depends how far you want to go back :blush:

Anyway with this regex we accept a ton of false positives, because the letter is always only one, and To be honest, I didn't understand with what logic the letter moves from the 1st position, then to the last and then to the 2nd and 3rd, etc.
In addition, in every province it is not certain that the letter has reached all the positions.

> Only support cars as of '94, which would result in a really simpler regex

You know what I think about it :smiley:.

Let me know your choises

Reply:
> Anyway with this regex we accept a ton of false positives, because the letter is always only one, and to be honest, I didn't understand the logic that moves the letter from the 1st position, then to the last and then to the 2nd and 3rd, etc.
> In addition, in every province it is not certain that the letter has reached all the positions.
> 

I don't know but I would say that is something similar to the hexadecimal approach maybe ? :thinking:  Whatever we opt for, there WILL be false positives or false negatives, depending on the choice.
 
> Let me know your choises

Just putting down a few considerations here, I don't pretend to give the final choice :wink: 

Anyway, given where our whole discussion lead to, I would opt for the simplest regexp - *that means documenting which plates **are and which are not** supported, though*.

Hope this helps ;)

Reply:
We should find a way to support plates from about the 80s since is difficult that an older car can still pass the overhaul. Maybe a new option for "deprecated plates" 😆. It would be useful for other countries too.

Reply:
We can reduce the number of false positives with something like:

`<city-codes>\s([A-Z]\d{5}|\d[A-Z]\d{4}|\d{2}[A-Z]\d{3}|\d{3}[A-Z]\d{2}…)`

Reply:
> We can reduce the number of false positives with something like:
> 
> `<city-codes>\s([A-Z]\d{5}|\d[A-Z]\d{4}|\d{2}[A-Z]\d{3}|\d{3}[A-Z]\d{2}…)`


In Milan, Rome and Turin (the most populouse provinces), the latest registered plates are [MI 01D000](https://www.targheitaliane.com/tabelle-targhe-1927/tabella-completa-1927/?pag=127&tipo=Autoveicolo&prov=MI), [Roma 5K0000](https://www.targheitaliane.com/tabelle-targhe-1927/tabella-completa-1927/?pag=115&tipo=Autoveicolo&prov=Roma) and [TO 50000W](https://www.targheitaliane.com/tabelle-targhe-1927/tabella-completa-1927/?pag=165&tipo=Autoveicolo&prov=TO)

So we can reduce even more with:

`<city-codes>\s([A-Z]\d{5}|\d[A-Z]\d{4}|\d{2}[A-D]\d{3}|\d{5}[A-Z])` 

and improve performances by ordering chronologically the conditions

`<city-codes>\s(\d{2}[A-D]\d{3}|\d[A-Z]\d{4}|\d{5}[A-Z]|[A-Z]\d{5} ...and if you want to supports the really oldest one... |\d{1,6})`

Reply:
Well for old plates, I'd rather test `[A-Z0-9]` for each... Maybe it' s not the most accurate, but it's a waaaay simpler regex ^^

Reply:
We need to come full circle.

A) Plates from '94
`/^[ABCDEFGHJKLMNPRSTVWXYZ]{2}\s?\d{3}\s?[ABCDEFGHJKLMNPRSTVWXYZ]{2}$/`

B) Plates from '68 (less accurate)
`/^ <plates-from-94> | <city-code>\s?[A-Z0-9]{6}`

C) Plates from '68 (more accurate) 
`/^ <plates-from-94> | <city-code>\s?(\d{2}[A-D]\d{3}|\d[A-Z]\d{4}|\d{5}[A-Z]|[A-Z]\d{5})`

D) Plates from '27 (less accurate) 
`/^ <plates-from-94> | <city-code>\s?[A-Z0-9]{1,6}`

E) Plates from '27 (more accurate) 
`/^ <plates-from-94> | <city-code>\s?(\d{2}[A-D]\d{3}|\d[A-Z]\d{4}|\d{5}[A-Z]|[A-Z]\d{5}|\d{1,6})`

I prefer A and E
Place your bets 😄

Reply:
I am for C🔝

Reply:
I would opt for B or D => the simplest ones :wink:

Reply:
Perfect!!! We have ourselves in a Mexican standoff 🤣😄

Reply:
@fedeci your second best is crucial

Reply:
I'd go with E, but splitting the regex in new and old plates. Nevertheless I am still not 100% convinced that cars with plates older than 70s are still circulating.

Reply:
> I'd go with E, but splitting the regex in new and old plates. Nevertheless I am still not 100% convinced that cars with plates older than 70s are still circulating.

@fedeci I totally agree with this :wink: Nevertheless, what about a website about old collection cars ? :thinking:

Reply:
@fasenderos What about the [idea of splitting old/new plates into 2 distinct vars](https://github.com/validatorjs/validator.js/pull/1677/files#r646186036) ?

Reply:
Hummm.. probably I'm missing something. 
Honestly I didn't know the differences between:

```
let oldPlates = '<old-plates-regex>';
let newPlates = '<new-plates-regex>';
'it-IT': str => (newPlates.test(str) || oldPlates.test(str))
```
and
```
'it-IT': str => /^(<new-plates-regex> | <old-plates-regex>)$/.test(str)
```
Doesn't the `|` (or) in the regex stop checking at first match?

Anyway that's fine to me, I'm just curious to know the differences

Reply:
Yes, but if any ambiguous sequence occurs it may lead the regex to be extremely slow.

Reply:
Hi guys, sorry for the delay, I've been a little busy lately. I just updated the regex with E, by splitting the regex in new and old plates.